tests/Amara/Varcon/HtmlTranslatorTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Amara\Varcon\Tests;
+
+use Amara\Varcon\HtmlTranslator;
+use Amara\Varcon\Translator;
+use Prophecy\Argument;
+
+class HtmlTranslatorTest extends \PHPUnit_Framework_TestCase
+{
+    public function testTranslatePreservesWhitespace()
+    {
+        $htmlTranslator = new HtmlTranslator();
+
+        $this->assertSame(
+            '<p>Colour <strong>pyjama</strong> паралелепипед</p>', // Tests UTF8 characters as well
+            $htmlTranslator->translate(
+                '<p>Color <strong>pajama</strong> паралелепипед</p>',
+                'A',
+                'B'
+            )
+        );
+    }
+
+    /**
+     * @dataProvider provideTranslate
+     *
+     * @param string $html
+     * @param string $translatedHtml
+     */
+    public function testTranslate($html, $translatedHtml)
+    {
+        $translator = $this->prophesize(Translator::class);
+        $translator->translate(
+            Argument::any(),
+            Argument::any(),
+            Argument::any(),
+            Argument::any()
+        )->willReturn('Translated'); // Keep in mind we ignore whitespace this way
```

Comment:
It might make this test better to have it return the text passed in appended with 'Translated' rather than always 'Translated'